src/pipeline/rackhd/docker/build.sh

```diff
@@ -0,0 +1,393 @@
+    --manifest ${local_manifest_file} \
+    --builddir ${dest_dir} \
+    --jobs 8 \
+    --force \
+    checkout \
+    packagerefs-commit
+    mv $local_manifest_file  ${dest_dir}
+}
+
+
+
+# calculate tag
+# input: repo directory with on-xxx src code inside it
+# input: RACKHD_CHANGELOG_VERSION in global var 
+# output: PKG_TAG as a global var
+tagCalculate() {
+    local repo=$1
+    local rackhd_changelog_version=$2
+    pushd $repo
+        #Get package version from debian/changelog
+        local CHANGELOG_VERSION=""
+
+        if [ -f "debian/changelog" ]; then
+            CHANGELOG_VERSION=$(dpkg-parsechangelog --show-field Version)
+        elif [ -f "debianstatic/$repo/changelog" ]; then
+            cp -rf debianstatic/$repo ./debian
+            CHANGELOG_VERSION=$(dpkg-parsechangelog --show-field Version)
+            rm -rf debian
+        else
+            CHANGELOG_VERSION=$rackhd_changelog_version
+        fi
+        local GIT_COMMIT_DATE=$(git show -s --pretty="format:%ci")
+        local DATE_STRING="$(date -d "$GIT_COMMIT_DATE" -u +"%Y%m%dUTC")"
+        local GIT_COMMIT_HASH=$(git show -s --pretty="format:%h")
+        PKG_TAG="$CHANGELOG_VERSION-$DATE_STRING-$GIT_COMMIT_HASH"
+    popd
+}
+
+buildOnCore() {
+    local commitstring_file=$1
+    local on_core_tag=$2
+    echo "[DEBUG]build on-core"
+    pushd on-core
+        addCommitString ${commitstring_file}
+        docker build -t rackhd/on-core${on_core_tag} .
+    popd
+}
+
+buildOnTasks() {
+    local commitstring_file=$1
+    local on_core_tag=$2
+    local on_tasks_tag=$3
+    echo "[DEBUG]build on-tasks"
+    pushd on-tasks
+        addCommitString ${commitstring_file}
+        sed -i "s/^FROM rackhd\/on-core:devel/FROM rackhd\/on-core${on_core_tag}/" Dockerfile
+        docker build -t rackhd/on-tasks${on_tasks_tag} .
+    popd
+}
+
+buildOnImagebuilder() {
+    local commitstring_file=$1
+    local repo="on-imagebuilder"
+    echo "[DEBUG]build on-imagebuilder"
+    PKG_TAG=""
+    tagCalculate $repo $RACKHD_CHANGELOG_VERSION
+    pushd $repo
+        addCommitString ${commitstring_file}
+        echo "Building rackhd/$repo:${PKG_TAG}"
+        cp Dockerfile ../Dockerfile.bak
+        cp ${commitstring_file} ./common/
+        docker build -t rackhd/files:${PKG_TAG} .
+    popd
+}
+
+buildCommon() {
+    local repo=$1
+    local commitstring_file=$2
+    local on_core_tag=$3
+    local on_tasks_tag=$4
+    echo "[DEBUG]build $repo"
+
+    if [ ! -d $repo ]; then
+        echo "Repo directory of $repo does not exist"
+        popd > /dev/null 2>&1
+        exit 1
+    fi
+    PKG_TAG=""
+    tagCalculate $repo $RACKHD_CHANGELOG_VERSION
+    pushd $repo
+        addCommitString ${commitstring_file}
+
+        echo "Building rackhd/$repo:${PKG_TAG}"
+        cp Dockerfile ../Dockerfile.bak
+        #Based on newly build upstream image to build
+        sed -i "s/^FROM rackhd\/on-core:devel/FROM rackhd\/on-core${on_core_tag}/" Dockerfile
```

Comment:
Sorry, there are some changes in master(https://github.com/RackHD/on-build-config/pull/313).
Please sync the changes to your PR~